lib/edge_impulse/CMakeLists.txt

```diff
@@ -0,0 +1,138 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+
+if(CONFIG_EDGE_IMPULSE_ZIP_PATH STREQUAL "")
+  message(FATAL_ERROR "Edge Impulse archive path must be specified.")
+endif()
+
+# Set full path for Edge Impulse archive and base directory
+set(EI_ZIP_PATH ${APPLICATION_SOURCE_DIR}/${CONFIG_EDGE_IMPULSE_ZIP_PATH})
+set(EI_BASE_DIR ${CMAKE_BINARY_DIR}/ei_src_unpacked)
+
+# Download Edge Impulse library (if needed)
+if(NOT CONFIG_EDGE_IMPULSE_DOWNLOAD_URI STREQUAL "")
+  if(NOT EXISTS ${EI_ZIP_PATH})
+    # Start download, use API key that was provided.
+    message(STATUS "Downloading Edge Impulse files, please wait...")
+
+    if(DEFINED EI_API_KEY_HEADER)
+      file(DOWNLOAD ${CONFIG_EDGE_IMPULSE_DOWNLOAD_URI} ${EI_ZIP_PATH}
+           SHOW_PROGRESS
+           LOG download_log
+           STATUS download_status
+           HTTPHEADER "Accept: application/zip"
+           HTTPHEADER ${EI_API_KEY_HEADER})
+    else()
+      file(DOWNLOAD ${CONFIG_EDGE_IMPULSE_DOWNLOAD_URI} ${EI_ZIP_PATH}
+           SHOW_PROGRESS
+           LOG download_log
+           STATUS download_status
+           HTTPHEADER "Accept: application/zip")
+    endif()
+
+    list(GET download_status 0 status_code)
+    if (status_code EQUAL 0)
+      message(STATUS "Successfully downloaded Edge Impulse archive.")
+    else()
+      file(REMOVE ${EI_ZIP_PATH})
+      message(FATAL_ERROR "Failed to download Edge Impulse files. Please make "
+               "sure that CONFIG_EDGE_IMPULSE_DOWNLOAD_URI is properly defined "
+               "and EI_API_KEY_HEADER is specified (if needed).")
+    endif()
+  else()
+    message(STATUS "Edge Impulse zip archive is already downloaded.")
+  endif()
+else()
+  if (NOT EXISTS ${EI_ZIP_PATH})
+    message(FATAL_ERROR "Specified Edge Impulse archive (${EI_ZIP_PATH}) does "
+             "not exist. Download the archive manually and place it under "
+             "defined path or use CONFIG_EDGE_IMPULSE_DOWNLOAD_URI Kconfig "
+             "option to let the build system download the library "
+             "automatically.")
+  else()
+    message(STATUS "Using Edge Impulse archive: ${EI_ZIP_PATH}.")
+  endif()
+endif()
+
+# Unzip Edge Impulse archive
+if (NOT EXISTS ${EI_BASE_DIR})
+  file(MAKE_DIRECTORY ${EI_BASE_DIR})
+endif()
+
+execute_process(
+  COMMAND ${CMAKE_COMMAND} -E tar x ${EI_ZIP_PATH}
```

Comment:
We could use the FILE ARCHVE_CREATE functionality introduced in CMake 3.18.5 here to make unzipping more universal (that would require switching to newer CMake version in NCS)

Reply:
@tejlmand, what do you think about it ?

Reply:
I added CMake required version to the file and switched to using the new API